app/controllers/application_controller.rb

```diff
@@ -77,4 +83,29 @@ def prepare_exception_notifier
   end
 
 
+  # When one of our URLS is cut and pasted in Facebook (FB), FB will actually
+  # _change_ the URL.  Then anyone in FB clicking on the URL will be taken
+  # to our system with a (possibly) invalid URL.
+  # Ex:  I want to share the list of dog groomers in Skäne (a region) with someone.
+  #   I do a search for all companies in Skäne with dog groomers.  I copy that resulting URL
+  #   and share it with my friends in FB.  One of my friends clicks on it but
+  #   because FB has mangled the URL, it no longer works and causes a 500 or 404 error.
+  #
+  # Specifically, FB may change  a value that should be an Array to a Hash.  this changes it back.
+  # This is important for the Ransack gem, which we use for searching and sorting in some views.
+  # (Search for the `.ransack` method in controllers.)
+  # Ransack expects the parameters with the 'q' key to have values that are Arrays, not Hashes.
```

Comment:
It's pretty clear that FB is converting our params to JSON, and then just serving that JSON right back to us without backing out the conversion.  This may or may not be worth mentioning here.